### PR TITLE
Getting experiential on a Quarkus project

### LINKED ISSUE #2975: Getting experiential on a Quarkus project
### Proposed title of the article

Getting experiential on a Quarkus project

### Introduction paragraph (2-3 paragraphs):

Quarkus is an open-source Java native framework useful for creating and managing Kubernetes native Java stack, microservices, and serverless environments which is optimized for the open JDK and the GraalVM runtime. It is a very effective platform for Kubernetes, cloud, and serverless environments.
In similarity to the Spring Boot framework, Quarkus is a combination of different Java libraries and standards, providing you a consolidation along with improvements on the JVM.
Although this framework is young compared with the language (over 20 years old), it has some promising features and results that make it catchy to developers.

In this tutorial, we will get started on Quarkus and see how we can have some hands-on Quarkus projects using different tools and software environments.

### Key takeaways

Coming to the tip of the article, we shall have the following to take with us:

- What is Quarkus
- Quarkus prerequisites
- Advantages and trade-offs of Quarkus framework
- Creation of a Quarkus application using different methods or tools such as from the [https://code.quarkus.io/](https://code.quarkus.io/) website, using Visual Studio Code and IntelliJ.
- Quarkus project structure
- Differences between the Quarkus framework and the Spring Boot framework.
- Testing Quarkus resources usage using JConsole.

### Pre-requisites

Some basics required for easy follow-up on this article include:

- A modern IDE that supports Java frameworks installed on your machine. This may either be Visual Studio Code or IntelliJ or any other. We shall focus on the two in this article.
- A stable internet connection for dependency downloads.
- JDK 11+ of any distribution installed. This is because Quarkus is not supported with lower versions than 11.
- Gradle or Apache Maven 3.8.1+ installed.
- Java programming skills (It will be a bit hard for a beginner to understand or catch up)

### References

- [https://developer.ibm.com/components/ibm-mq/tutorials/mq-running-ibm-mq-apps-on-quarkus-and-graalvm-using-qpid-amqp-jms-classes/](https://developer.ibm.com/components/ibm-mq/tutorials/mq-running-ibm-mq-apps-on-quarkus-and-graalvm-using-qpid-amqp-jms-classes/)

### Templates to use as guides

- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by RisoriTofa**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**

This fixes #2975.      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by RisoriTofa** [comment]
This fixes #2975

File: content/articles/getting-experiential-on-a-quarkus-project/index.md [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+
+> **[GraalVM](https://www.graalvm.org/)**: a high-performance JDK distribution.
+> Is a high-performance Java Development Kit (JDK) distribution built to quicken the application executions written in Java and other JVM languages.
+> It provides other runtimes for Python, JavaScript, Ruby, and some other popular languages.
+> Availability of polyglot capabilities makes it possible to mix multiple programming languages in a single application while removing call costs of any foreign language.
+
+Lastly, as for Quarkus and Microprofile relationship, Microprofile is used to do microservices specification in the Java Enterprise environment.
+Quarkus on the other hand is an implementation of Microprofile which is well-built based on the given specifications.
+This makes them both compatible with each other.
+If you have worked with Microprofile, Quarkus is going to be simpler for you to work with.
```
Comment (jerimkaura) [inline content/articles/getting-experiential-on-a-quarkus-project/index.md]: would you please consider doing the comparison in tabular or list form to bring out more clarity?

File: content/articles/getting-experiential-on-a-quarkus-project/index.md [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+Just head over to the site. It looks like this:
+
+![code.quarkus.io](code.quarkus.io.png "The main code.quarkus.io page")
+_(Screenshot by Author)_
+
+You can see that you can rename the _Group_, _Artifact_, and the _Build Tool_.
+You can search and select the needed dependencies before generating your project zip file.
+Dependencies can also be added using the Quarkus CLI, as you will later see on.
+You may also add more configurations such as the SNAPSHOT version and availability of Starter code by clicking on the 'CONFIGURE MORE OPTIONS'.
+
```
Comment (jerimkaura) [inline content/articles/getting-experiential-on-a-quarkus-project/index.md:131]: List our what tone can do in point form

**Review by jerimkaura — CHANGES_REQUESTED** [review]
@RisoriTofa  Hello, a nice piece here:rocket: 

When describing a process, please put it in point form to avoid lengthy paragraphs. Do that for all the steps you describe to make the article easier to read.

Secondly, eliminate lengthy paragraphs by communicating only a single idea in one paragraph.

Make the changes above then we proceed to tech review. Otherwise, your article is clean and easy to follow.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by RisoriTofa** [comment]
> @RisoriTofa Hello, a nice piece here
> 
> When describing a process, please put it in point form to avoid lengthy paragraphs. Do that for all the steps you describe to make the article easier to read.
> 
> Secondly, eliminate lengthy paragraphs by communicating only a single idea in one paragraph.
> 
> Make the changes above then we proceed to tech review. Otherwise, your article is clean and easy to follow.

Ok, and thanks!

File: content/articles/getting-experiential-on-a-quarkus-project/index.md (hunk 1 above)
Comment (RisoriTofa) [reply content/articles/getting-experiential-on-a-quarkus-project/index.md]: i've decided to use the list form :+1:

Comment (RisoriTofa) [reply content/articles/getting-experiential-on-a-quarkus-project/index.md]: Done!

File: content/articles/getting-experiential-on-a-quarkus-project/index.md (hunk 2 above)
Comment (RisoriTofa) [reply content/articles/getting-experiential-on-a-quarkus-project/index.md:131]: Ok!

Comment (RisoriTofa) [reply content/articles/getting-experiential-on-a-quarkus-project/index.md:131]: Done!

**Review by jerimkaura — APPROVED** [review]

**Comment by jerimkaura** [comment]
@RisoriTofa, Quite a detailed article :rocket: check on the corrections made to improve your articles.

**Comment by Kayere** [comment]
Great article @RisoriTofa.
Thank you for the review @jerimkaura.

**Comment by Kayere** [comment]
Great article @RisoriTofa.
Thank you for the review @jerimkaura.

**Pull request merged**